exporter_test.go

```diff
@@ -0,0 +1,436 @@
+		ctx:      ctx,
+		monitor:  monitor,
+		callback: callback,
+	})
+	
+	if m.streamErr != nil {
+		return m.streamErr
+	}
+	
+	// Simulate successful streaming with mock data
+	realtimeUpdate := &realtime.RealtimeUpdate{
+		W: 100,
+		Hz: 60,
+		Voltage: []float32{120.5, 119.8},
+		Devices: []realtime.Device{
+			{ID: "device1", W: 50},
+			{ID: "device2", W: 30},
+		},
+	}
+	
+	deviceStates := &realtime.DeviceStates{
+		States: []realtime.DeviceState{
+			{DeviceID: "device1", Mode: "active", State: "online"},
+			{DeviceID: "device2", Mode: "inactive", State: "offline"},
+		},
+	}
+	
+	// Call the callback with mock data
+	if err := callback(ctx, realtimeUpdate); err != nil {
+		if err == realtime.Stop {
+			return nil // Stop is expected, not an error
+		}
+		return err
+	}
+	
+	if err := callback(ctx, deviceStates); err != nil {
+		if err == realtime.Stop {
+			return nil // Stop is expected, not an error
+		}
+		return err
+	}
+	
+	return nil
+}
+
+func (m *mockClient) GetMonitors() []sense.Monitor {
+	return append([]sense.Monitor{}, m.monitors...)
+}
+
+func TestNewCollector(t *testing.T) {
+	tests := []struct {
+		name      string
+		ctx       context.Context
+		client    exporter.Client
+		monitorID int
+		timeout   time.Duration
+	}{
+		{
+			name:      "creates collector with valid parameters",
+			ctx:       context.Background(),
+			client:    &mockClient{userID: 123, accountID: 456},
+			monitorID: 789,
+			timeout:   5 * time.Second,
+		},
+		{
+			name:      "creates collector with zero timeout",
+			ctx:       context.Background(),
+			client:    &mockClient{userID: 123, accountID: 456},
+			monitorID: 789,
+			timeout:   0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			collector := exporter.NewCollector(tt.ctx, tt.client, tt.monitorID, tt.timeout)
+			
+			if collector == nil {
+				t.Fatal("NewCollector returned nil")
+			}
+			
+			// We can't directly test unexported fields, but we can test the behavior
+			// by calling methods that depend on these fields
+		})
+	}
+}
+
+func TestCollectorDescribe(t *testing.T) {
+	client := &mockClient{userID: 123, accountID: 456}
+	collector := exporter.NewCollector(context.Background(), client, 789, time.Second)
+	
+	ch := make(chan *prometheus.Desc, 10)
+	
+	go func() {
+		collector.Describe(ch)
+		close(ch)
+	}()
+	
+	// Collect all descriptors
+	var descs []*prometheus.Desc
+	for desc := range ch {
+		descs = append(descs, desc)
+	}
+	
+	// Verify we got the expected number of descriptors
+	expectedCount := 8 // upDesc, scrapeTimeDesc, deviceWattsDesc, voltsDesc, wattsDesc, hzDesc, activeDesc, onlineDesc
+	if len(descs) != expectedCount {
+		t.Errorf("Expected %d descriptors, got %d", expectedCount, len(descs))
+	}
+	
+	// Verify all descriptors are not nil
+	for i, desc := range descs {
+		if desc == nil {
+			t.Errorf("Descriptor at index %d is nil", i)
+		}
+	}
+}
+
+func TestCollectorCollect(t *testing.T) {
```

Comment:
Is this test redundant with TestCollectorCollectMetricsContent?  If so, it seems like TestCollectorCollectMetricsContent is more comprehensive.  Should they be consolidated?  Ignore my original instructions for table-driven tests if there's no benefit to trying to do so.